common/app/views/support/GetClasses.scala

```diff
@@ -157,4 +167,28 @@ object GetClasses {
         "fc-container--video-no-fill-sides" -> frontId.contains("video"),
       ) collect { case (kls, true) => kls }: _*,
     )
+
+  def paletteClass(container: Container, metadata: Seq[Metadata]): Option[String] =
+    container match {
+      case Fixed(_) => primaryPaletteClass(metadata)
+      case _        => None
+    }
+
+  private val paletteClassesByMetadataTag: Map[Metadata, String] = Map(
+    LongRunningPalette -> "fc-container--long-running-palette",
+    SombrePalette -> "fc-container--sombre-palette",
+    InvestigationPalette -> "fc-container--investigation-palette",
+    BreakingPalette -> "fc-container--breaking-palette",
+    EventPalette -> "fc-container--event-palette",
+    EventAltPalette -> "fc-container--event-alt-palette",
+  )
+
+  private def primaryPaletteClass(metadata: Seq[Metadata]): Option[String] = {
+    primaryPaletteTag(metadata).map(tag => paletteClassesByMetadataTag(tag))
+  }
+
+  private def primaryPaletteTag(metadata: Seq[Metadata]): Option[Metadata] = {
+    val paletteMetadataTags = paletteClassesByMetadataTag.keySet
+    metadata.find(tag => paletteMetadataTags.contains(tag))
```

Comment:
Here we are selecting up to one matching Metadata palette tag

Reply:
nitpick, but I was thinking moving this method above `primaryPaletteClass` might read more naturally?